File: SpringBootServer/carcar-backend/src/main/java/at/carcar/carcarbackend/Report/DamageReport.java
package at.carcar.carcarbackend.Report;
import at.carcar.carcarbackend.Car.Car;
import at.carcar.carcarbackend.Trip.Trip;
import at.carcar.carcarbackend.User.User;
import jakarta.persistence.ElementCollection;
import jakarta.persistence.Entity;

import java.util.Date;
import java.util.List;
@Entity
public class DamageReport extends Report {
    private String damageDetails;
    @ElementCollection
    private List<String> damagePhotos;

    public DamageReport(User author_user, Date date, String description, Trip trip, String damageDetails, Car car) {
        super(author_user, date, description, trip, car);
        this.damageDetails = damageDetails;
    }

    public DamageReport(User author_user, Date date, String description, String damageDetails, Car car) {
        super(author_user, date, description, car);
        this.damageDetails = damageDetails;
    }

    public DamageReport() {
    }

    public String getDamageDetails() {
        return damageDetails;
    }

    public void setDamageDetails(String damageDetails) {
        this.damageDetails = damageDetails;
    }

    public List<String> getDamagePhotos() {
        return damagePhotos;
    }

    public void setDamagePhotos(List<String> damagePhotos) {
        this.damagePhotos = damagePhotos;
    }

}

File: AndroidApp/app/src/main/java/com/example/carcarapplication/data_classes/DamageReport.kt
package com.example.carcarapplication.data_classes

import com.squareup.moshi.JsonClass
import java.io.Serializable
import java.time.LocalDateTime

@JsonClass(generateAdapter = true)
data class DamageReport(
    val damageDetails: String = "",
    //val damagePhotos: List<String?>, // Use String (e.g., URLs or file paths) instead of Image
    override val reportID: Long?  = 0,
    override val author_user: User? = null,
    override val date: LocalDateTime? = null,
    override val description: String = "",
    // override val changeLog: List<String>,
    override val trip: Trip? = null,
    override val car: Car? = null
) : Report(reportID, author_user, date, description, trip, car), Serializable

